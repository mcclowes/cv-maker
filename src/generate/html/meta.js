const meta = (config) => {
  if(!config || !config.name) {
    return `<title>CV</title>
    <meta name="description" content="CV made with cv-maker">

    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no">

    <link rel="icon" href="/favicon.ico" />`
  }

  return `
    <title>${config.name} CV</title>
    <meta name="description" content="${config.description}">

    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no">

    <link rel="icon" href="/favicon.ico" />

    <!--  Social tags -->
    <!--  Essential META Tags -->
    <meta property="og:title" content="The CV of ${config.name}">
    <meta property="og:description" content="${config.description}">
    <meta property="og:image" content="${config.previewImage}">
    <meta property="og:url" content="${config.url}">
    <meta name="twitter:card" content="summary_large_image">


    <!--  Non-Essential, But Recommended -->
    <meta property="og:site_name" content="The CV of ${config.name}">
    <meta name="twitter:image:alt" content="${config.previewImageText}">


    <!--  Non-Essential, But Required for Analytics -->
    <meta name="twitter:site" content="${config.twitterUsername}">
  `
}

export default meta;