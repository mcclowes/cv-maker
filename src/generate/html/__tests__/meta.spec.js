import meta from "../meta";

const CONFIG = {
  name: "Joe Bloggs",
  description: "Enter a summary of you here.",
  previewImage: "https://cv.joebloggs.com/preview.png",
  previewImageText: "Joe Bloggs CV",
  url: "https://cv.joebloggs.com/",
  twitterUsername: "@joebloggs",
}

describe("meta", () => {
  it("generates correct meta html", () => {
    expect(
      meta(CONFIG)
    ).toMatchSnapshot();
  });
});
