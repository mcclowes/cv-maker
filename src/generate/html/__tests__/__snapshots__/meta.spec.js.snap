// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta generates correct meta html 1`] = `
"
    <title>Joe Bloggs CV</title>
    <meta name=\\"description\\" content=\\"Enter a summary of you here.\\">

    <meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no\\">

    <link rel=\\"icon\\" href=\\"/favicon.ico\\" />

    <!--  Social tags -->
    <!--  Essential META Tags -->
    <meta property=\\"og:title\\" content=\\"The CV of Joe Bloggs\\">
    <meta property=\\"og:description\\" content=\\"Enter a summary of you here.\\">
    <meta property=\\"og:image\\" content=\\"https://cv.joebloggs.com/preview.png\\">
    <meta property=\\"og:url\\" content=\\"https://cv.joebloggs.com/\\">
    <meta name=\\"twitter:card\\" content=\\"summary_large_image\\">


    <!--  Non-Essential, But Recommended -->
    <meta property=\\"og:site_name\\" content=\\"The CV of Joe Bloggs\\">
    <meta name=\\"twitter:image:alt\\" content=\\"Joe Bloggs CV\\">


    <!--  Non-Essential, But Required for Analytics -->
    <meta name=\\"twitter:site\\" content=\\"@joebloggs\\">
  "
`;
